pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -989,26 +1163,62 @@ def __init__(
         if not control_values:
             control_values = "1" * len(control_wires)
 
-        control_int = self._parse_control_values(control_wires, control_values)
         self.control_values = control_values
-
-        self._padding_left = control_int * 2
-        self._padding_right = 2 ** len(wires) - 2 - self._padding_left
-        self._CX = None
+        self.control_int = self._parse_control_values(self.control_wires, self.control_values)
+        self._cx = None
 
         super().__init__(*params, wires=wires, do_queue=do_queue)
 
     @property
     def num_params(self):
         return 0
 
-    def _matrix(self, *params):
-        if self._CX is None:
-            self._CX = block_diag(
-                np.eye(self._padding_left), PauliX.matrix, np.eye(self._padding_right)
-            )
+    @staticmethod
+    def compute_matrix(control_int, n_wires):
+        """Canonical matrix representation of the MultiControlledX operator.
+
+        **Examples:**
+
+        >>> qml.MultiControlledX.compute_matrix(0, 2)
+        [[1, 0, 0, 0],
+        [0, 1, 0, 0],
+        [0, 0, 0, 1],
+        [0, 0, 1, 0]]
+        >>> qml.MultiControlledX.compute_matrix(1, 2)
+        [[0. 1. 0. 0.]
+         [1. 0. 0. 0.]
+         [0. 0. 1. 0.]
+         [0. 0. 0. 1.]]
+
+        Args:
+           control_int (int): index of wire that the X gate acts on
+           n_wires (int): total number of wires
 
-        return self._CX
+        Returns:
+           tensor-like or None: matrix representation
+        """
+        padding_left = control_int * 2
+        padding_right = 2 ** n_wires - 2 - padding_left
+        cx = block_diag(np.eye(padding_left), PauliX.compute_matrix(), np.eye(padding_right))
+        return cx
+
+    def matrix(self, wire_order=None):
+
+        if self._cx is None:
+            # store matrix, so we do not have to do this each time
+            self._cx = self.compute_matrix(self.control_int, len(self.wires))
+
+        canonical_matrix = self._cx
```

Comment:
I think this might suffer from the same problem as above; the stored canonical matrix won't be returned if `control_int` or the wires change.

Which I think is solved using the cache decorator approach instead